### PR TITLE
move to Dask array + first working version of the multiformer

### PR DESCRIPTION
**Pull request by iluise**
adding the following features: 

1) Dask arrays in `multifield_dala_sampler`
2) faster way to retrieve the `normalizers` when lats-lons are in increasing order

3) fix embed/embeds mismatch when loading checkpoints
4) running multiformer configuration (multiples changes in `train_multi.py`), also from single field checkpoints
5) a bit of cleaning

### CODE REVIEW

**Comment by kacpnowak** [comment]
Would it be possible to use xarray to open zarr files? It works much better with timestamps and it's compatible with dask

**Comment by clessig** [comment]
> Would it be possible to use xarray to open zarr files? It works much better with timestamps and it's compatible with dask

Could you reply a clear example where the benefit would be?

Dask is significantly faster than the native python zarr library. What is the performance of xarray?

**Comment by kacpnowak** [comment]
I didn't mean to use xarray instead of dask. I mean using it: here

`self.ds = zarr.open( file_path)`
  
In case of xarray it's just:

`self.ds = xr.open_zarr( file_path)` 

In such case for datetime type casting becomes uncessary, eg. `self.year_base = self.ds['time'][0].astype(datetime)` as one can simply use `self.ds.index['time'][0]`

Additionally it better decodes timestamps. In case of FESOM data when timestamps are in daily resolution, I had to manually do conversion from int. Xarray figured it out on it's own

**Comment by clessig** [comment]
> for datetime type casting becomes uncessary, eg. `self.year_base = self.ds['time'][0].astype(datetime)` as one can simply use `self.ds.index['time'][0]`
> 
> Additionally it better decodes timestamps. In case of FESOM data when timestamps are in daily resolution, I had to manually do conversion from int. Xarray figured it out on it's own

But by using xarray we would lose the performance advantage of dask, I assume. I would want to see that it's performance neutral before considering to use xarray instead of dask.

**Comment by mlangguth89** [comment]
xarray has a bulit-in support for dask:
https://docs.xarray.dev/en/latest/user-guide/dask.html

**Comment by kacpnowak** [comment]
@clessig There's no performance penalty for using Xarray. here is quite extensive comparison between zarr and xarray performance: https://github.com/pydata/xarray/issues/9111#issuecomment-2164233488

**Comment by clessig** [comment]
The question is wrt to dask (that might parallelize things etc) and not the native zarr library. Could you run with dask and xarray and num_worker_loaders=0 and see how many sampler/sec we get. Thanks!

**Comment by kacpnowak** [comment]
I'm on vacation so I can do it next week

**Comment by clessig** [comment]
Where do we stand here? Ready to merge?

@kacpnowak : did you check the performance of xarray?

**Comment by kacpnowak** [comment]
So far it's too difficult to make xarray only version. On developer meeting with @mlangguth89 we've talked about doing during hackaton in November, but I think it would be issue/branch

File: slurm_atmorep.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/bin/bash -x
 #SBATCH --account=ehpc03
-#SBATCH --time=0-24:00:00
-#SBATCH --nodes=1
-#SBATCH --ntasks-per-node=4
-#SBATCH --cpus-per-task=20
-#SBATCH --gres=gpu:4
+#SBATCH --time=0-71:59:59
```
Comment (clessig) [inline slurm_atmorep.sh:3]: Can we not merge these mn5 specific things

File: slurm_atmorep_evaluate.sh [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!/bin/bash -x
 #SBATCH --account=ehpc03
-#SBATCH --time=0-3:30:00
+#SBATCH --time=0-4:30:00
```
Comment (clessig) [inline slurm_atmorep_evaluate.sh:3]: Can we not merge the mn5 specific things

File: atmorep/core/train.py [hunk 3]
```diff
@@ -31,7 +31,7 @@
 ####################################################################################################
 def train_continue( wandb_id, epoch, Trainer, epoch_continue = -1) :
 
-  num_accs_per_task = int( 4 / int( os.environ.get('SLURM_TASKS_PER_NODE', '1')[0] ))
+  num_accs_per_task =  1 #int( 4 / int( os.environ.get('SLURM_TASKS_PER_NODE', '1')[0] ))
```
Comment (clessig) [inline atmorep/core/train.py:34]: This needs to be adapted to the changes I did to init torch that removed the need for num_accs_per_task.

Comment (clessig) [reply atmorep/core/train.py:34]: Since it's my fault, I can do it. But fine if you take care of it.

**Review by clessig — APPROVED** [review]
Some minor things only--would be great if they could be fixed before merging.

File: atmorep/core/train.py (hunk 3 above)
Comment (iluise) [reply atmorep/core/train.py:34]: ah yes, you're right, will do it no problem!

Comment (iluise) [reply atmorep/core/train.py]: Hi, I added it and tested to both train_continue and evaluate. Please take a look when you have a minute

**Review by clessig — APPROVED** [review]

**Pull request merged**